clouddriver-cloudrun/src/main/java/com/netflix/spinnaker/clouddriver/cloudrun/op/manifest/CloudrunDeployManifestOperation.java

```diff
@@ -186,6 +214,101 @@ private List<String> insertSpinnakerAppNameServiceNameVersionName(
         .collect(Collectors.toList());
   }
 
+  private List<String> bindArtifacts(
+      List<CloudrunService> configFiles, Collection<Artifact> artifacts) {
+
+    return configFiles.stream()
+        .map(
+            (configFile) -> {
+              try {
+                ObjectMapper yamlReader = new ObjectMapper(new YAMLFactory());
+                CloudrunService yamlObj = configFile;
+                if (yamlObj != null) {
+                  if (yamlObj.getMetadata() != null) {
+                    LinkedHashMap<String, Object> metadataMap = yamlObj.getMetadata();
+                    LinkedHashMap<String, Object> specMap = yamlObj.getSpec();
+                    if (metadataMap != null && specMap != null) {
+                      if (metadataMap.get("annotations") != null
+                          && ((LinkedHashMap<String, Object>) specMap.get("template")).get("spec")
+                              != null) {
+                        LinkedHashMap<String, Object> specSpecMap =
+                            (LinkedHashMap<String, Object>)
+                                ((LinkedHashMap<String, Object>) specMap.get("template"))
+                                    .get("spec");
+                        LinkedHashMap<String, Object> annotationsMap =
+                            (LinkedHashMap<String, Object>) metadataMap.get("annotations");
+                        bindTheRequiredArtifact(annotationsMap, specSpecMap, artifacts);
```

Comment:
Same comment here re: POJOs.

Reply:
@mattgogerly I will look into it. Might take some time.

Reply:
@mattgogerly ...done ...Added changes...Pls review

Reply:
@mattgogerly ..Please can you review the changes

Reply:
@mattgogerly Please can you review the changes?

Reply:
@mattgogerly  Please can you review the changes?

Reply:
@mattgogerly Please can you review the changes?

Reply:
@mattgogerly Please can you review the changes?